test/ruby/test_method.rb

```diff
@@ -472,6 +472,19 @@ def o.foo; end; line_no = __LINE__
     o.singleton_class
     m4 = o.method(:bar)
     assert_equal("#<Method: #{c4.inspect}(#{c.inspect})#bar(foo)() #{__FILE__}:#{line_no}>", m4.inspect, bug15608)
+
+    bug17428 = '[ruby-core:101635] [Bug #17428]'
+    c4 = Class.new(c)
+    c4.class_eval { alias bar foo }
+    o = c4.new
+    o.singleton_class
+    m4 = o.method(:bar)
+    assert_equal("#<Method: String(Module).prepend(*)>", String.method(:prepend).inspect, bug17428)
+
+    c5 = Class.new(String)
+    c5.singleton_class.class_eval{alias prep prepend}
+    c6 = Class.new(c5)
+    assert_equal("#<Method: #{c6.inspect}(#{c5.inspect}).prep(prepend)(*)>", c6.method(:prep).inspect, bug17428)
```

Comment:
Could you add a spec for this, in the spirit of https://bugs.ruby-lang.org/issues/15608 trying to clarify what's the right Method#inspect output?

Reply:
I added this spec in TruffleRuby's spec/ruby copy, and will sync it with other ruby/spec copies at the end of the month.
So just a MRI test is fine, or you can also apply that diff if you want.
```diff
diff --git a/spec/ruby/core/method/shared/to_s.rb b/spec/ruby/core/method/shared/to_s.rb
index 1fbee870d6..0192cee867 100644
--- a/spec/ruby/core/method/shared/to_s.rb
+++ b/spec/ruby/core/method/shared/to_s.rb
@@ -72,4 +72,12 @@ describe :method_to_s, shared: true do
     @string = @m.send(@method).sub(/0x\h+/, '0xXXXXXX')
     @string.should.start_with? "#<Method: #<MethodSpecs::MySub:0xXXXXXX>.bar"
   end
+
+  ruby_version_is '2.7' do
+    ruby_bug '#17428', '2.7'...'3.0' do
+      it "shows the metaclass and the owner for a Module instance method retrieved from a class" do
+        String.method(:include).inspect.should.start_with?("#<Method: #<Class:String>(Module)#include")
+      end
+    end
+  end
 end
```